### PR TITLE
#51 - Added BearerAuthenticator

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #51 
Added `BearerAuthenticator`

### CODE REVIEW

File: src/test/java/com/artipie/http/client/auth/BearerAuthenticatorTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        );
+        MatcherAssert.assertThat(
+            "Port is correct",
+            fake.capturedPort(),
+            new IsEqual<>(321)
+        );
+        MatcherAssert.assertThat(
+            "Path is correct",
+            path.get(),
+            new IsEqual<>("/get_token")
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/http/client/auth/BearerAuthenticatorTest.java:86]: @olegmoz I'd suggest to extract all the items being checked here into variables: host, post and path and use them in `WwwAuthenticate` ctor and in the assertions

File: src/test/java/com/artipie/http/client/auth/BearerAuthenticatorTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        );
+    }
+
+    @Test
+    void shouldProduceBearerHeaderUsingTokenFormat() {
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/http/client/auth/BearerAuthenticatorTest.java:91]: @olegmoz and in this method I'd suggest to extract `mF_9.B5f-4.1JqM` into variable

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz two suggestions, please, take a look

**Comment by olegmoz** [comment]
@olenagerasimova thanks, made the improvements

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks, looks good to merge

**Pull request merged**